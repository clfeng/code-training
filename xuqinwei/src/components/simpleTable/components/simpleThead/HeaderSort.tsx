import { defineComponent, toRefs } from 'vue'

import { ColumnSortOrder } from '@/types/table'

export default defineComponent({
  name: 'SimpleTableHeaderCell',
  emits: ['sort'],
  props: {
    column: {
      type: Object,
      default: {}
    }
  },
  setup(props, { emit }) {
    let { column } = toRefs(props)

    let onChangeSort = (event: MouseEvent, order) => {
      event.stopPropagation()
      console.log('column.value', column.value);

      emit('sort', column, order)
    }
    return () => {
      return (
        <div class="table__sort">
          {
            /* 升序按钮 */
            column.value.sortOrders?.asc && <span class="asc" onClick={(event: MouseEvent) => onChangeSort(event, ColumnSortOrder.asc)}></span>
          }
          {
            /* 降序按钮 */
            column.value.sortOrders?.desc && <span class="desc" onClick={(event: MouseEvent) => onChangeSort(event, ColumnSortOrder.desc)}></span>
          }
        </div>
      )
    }
  },
})