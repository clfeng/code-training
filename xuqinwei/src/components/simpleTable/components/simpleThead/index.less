
.head-th{
  position: relative;
}
.asc{
  &::before{
    content: '^';
    position: absolute;
    top: -1%;
    right: 40%;
  }
}

.desc{
  &::after{
    content: '^';
    position: absolute;
    top: 1%;
    right: 40%;
    transform: rotateX(180deg);
  }
}
