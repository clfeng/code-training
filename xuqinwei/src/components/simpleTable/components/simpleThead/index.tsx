import { defineComponent } from 'vue'
import './index.less'

import HeaderSort from './HeaderSort'

export default defineComponent({
  name: 'simpleThead',

  emits: ['sort'],
  components: {
    HeaderSort
  },
  props: {
    dataColumns: {
      type: Array,
      default: []
    },
  },
  setup(props, { emit }) {
    function headClick(prop, order) {
      emit('sort', prop, order)

    }
    return () => {
      return <tr class="head-tr">
        {
          props.dataColumns.map(item => {
            return <th class="head-th">
              {item.label}
              {
                item.isSort && <header-sort column={item} onSort={headClick} />
              }
            </th>
          })
        }
      </tr>
    }
  },
})
