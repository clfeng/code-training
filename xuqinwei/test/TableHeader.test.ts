import { shallowMount, mount } from "@vue/test-utils";
import { expect, test, vi } from "vitest";
import simpleThead from '../src/components/simpleTable/components/simpleThead'
// import { columns } from './mockData'
let columns = [
  {
    dataIndex: 1,
    isSort: false,
    label: "姓名",
    order: false,
    prop: "name",
    sortOrders: [],
    width: 80
  },
  {
    dataIndex: 2,
    isSort: false,
    label: "性别",
    order: [],
    prop: "name",
    sortOrders: [],
    width: 80
  },
  {
    dataIndex: 3,
    isSort: true,
    label: "年龄",
    order: false,
    prop: "name",
    sortOrders: ['normal', 'asc', 'desc'],
    width: 80
  },
  {
    dataIndex: 4,
    isSort: false,
    label: "地址",
    order: false,
    prop: "name",
    sortOrders: [],
    width: 200
  },
]
test("table header", async () => {
  const headClick = vi.fn()
  const wrapper = shallowMount(simpleThead, {
    props: {
      dataColumns: columns
    },
    setup() {
      return {
        // headClick
      }
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
  await wrapper.find('.head-tr').findAll('.head-th')[3].trigger('click');
});

