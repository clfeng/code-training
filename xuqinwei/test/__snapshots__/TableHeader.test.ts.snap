// Vitest Snapshot v1

exports[`mount component 1`] = `"<colgroup></colgroup>"`;

exports[`table header 1`] = `""`;

exports[`xxx 1`] = `""`;
